joblib/test/test_parallel.py

```diff
@@ -152,6 +152,18 @@ def test_effective_n_jobs():
     assert effective_n_jobs() > 0
 
 
+def test_effective_n_jobs_None():
+    # check the number of effective jobs when `n_jobs=None`
+    # non-regression test for https://github.com/joblib/joblib/issues/984
+    backend_n_jobs = 3
+    with parallel_backend("threading", n_jobs=backend_n_jobs):
```

Comment:
It would be nice to also add a test that this work when n_jobs is not set in `parallel_backend` or set to `None`.

Reply:
When `n_jobs` is not set, it will default to `n_jobs=-1` and in this case it will always be the number of CPU counts.

Reply:
Yes and we cannot change this without breaking backward compat.